src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -122,6 +133,127 @@ public static Set<Tag> parseTags(Collection<String> tags) throws ParseException
         return tagSet;
     }
 
+    /**
+     * Parses a task description to determine the task type and details.
+     *
+     * @throws ParseException if the task format is invalid.
+     */
+    public static Task parseTask(String taskDescription) throws ParseException {
+        requireNonNull(taskDescription);
+        ParsedTask parsedTask = parseTaskTypeAndDetails(taskDescription);
+        String taskType = parsedTask.getTaskType();
+        String taskDetails = parsedTask.getTaskDetails();
+
+        switch (taskType) {
+        case "todo":
+            return parseTodoTask(taskDetails);
+        case "deadline":
+            return parseDeadlineTask(taskDetails);
+        case "event":
+            return parseEventTask(taskDetails);
+        default:
+            throw new ParseException(String.format(Messages.MESSAGE_INVALID_TASK_TYPE, taskType));
+        }
+    }
+
+    /**
+     * Parses {@code Collection<String> tags} into a {@code Set<Tag>}.
+     */
+    public static Set<Task> parseTasks(List<String> taskDescriptions) throws ParseException {
+        final Set<Task> taskSet = new HashSet<>();
+        for (String taskDescription : taskDescriptions) {
+            taskSet.add(parseTask(taskDescription));
+        }
+        return taskSet;
+    }
+
+    /**
+     * Parses the task description to extract the task type and task details.
+     *
+     * @throws ParseException if the format is invalid.
+     */
+    private static ParsedTask parseTaskTypeAndDetails(String taskDescription) throws ParseException {
+        requireNonNull(taskDescription);
+        String[] tokens = taskDescription.split("\\s+", 2);
+
+        if (tokens.length < 2) {
+            throw new ParseException(Messages.MESSAGE_INCOMPLETE_TASK_DESCRIPTION);
+        }
+
+        String taskType = tokens[0].toLowerCase();
+        String taskDetails = tokens[1].trim();
+
+        return new ParsedTask(taskType, taskDetails);
+    }
+
+
+    private static Todo parseTodoTask(String taskDetails) {
```

Comment:
Since the description of parserUtil is "Contains utility methods used for parsing strings in the various *Parser classes." it should not be limited to Person parsing methods, but I will add a separator as you have suggested, for person, task and wedding parsing methods